src/engraving/libmscore/barline.cpp

```diff
@@ -1150,17 +1167,24 @@ void BarLine::endEditDrag(EditData& ed)
                 b->setParent(s);
                 score()->undoAddElement(b);
             }
+            breakLast = b->spanTo();
             b->undoChangeProperty(Pid::BARLINE_SPAN, true);
         }
-        BarLine* b = toBarLine(s->element(staffIdx2 * VOICES));
-        if (b) {
-            b->undoChangeProperty(Pid::BARLINE_SPAN, false);
+        if (breakLast) {
+            BarLine* b = toBarLine(s->element(staffIdx2 * VOICES));
+            if (b) {
+                b->undoChangeProperty(Pid::BARLINE_SPAN, false);
+            }
         }
     } else {
+        bool breakLast = false;
         for (int staffIdx = staffIdx1; staffIdx < staffIdx2; ++staffIdx) {
+            breakLast = score()->staff(staffIdx)->barLineSpan();
             score()->staff(staffIdx)->undoChangeProperty(Pid::STAFF_BARLINE_SPAN, true);
         }
-        score()->staff(staffIdx2)->undoChangeProperty(Pid::STAFF_BARLINE_SPAN, false);
+        if (breakLast) {
+            score()->staff(staffIdx2)->undoChangeProperty(Pid::STAFF_BARLINE_SPAN, false);
```

Comment:
came out of discussions with @Tantacrul  and @MarcSabatella around when barlines should be broken after dragging handle down, essentially, they shouldn't be if barline was already broken before the staff you drag the barline onto.